packages/server-core/src/social/location/location.hooks.ts

```diff
@@ -45,72 +63,235 @@ import {
   locationResolver
 } from './location.resolvers'
 
-const applyLocationSettingSort = async (context: HookContext, next: NextFunction) => {
-  await next() // Read more about execution of hooks: https://github.com/feathersjs/hooks#flow-control-with-multiple-hooks
-
+const sortByLocationSetting = async (context: HookContext<LocationService>) => {
   const hasLocationSettingSort =
     context.params.query &&
     context.params.query.$sort &&
     Object.keys(context.params.query.$sort).some((item) => locationSettingSorts.includes(item))
 
-  if (hasLocationSettingSort) {
-    const { dispatch } = context
-
+  if (hasLocationSettingSort && context.params.query && context.params.query.$sort) {
     for (const sort of Object.keys(context.params.query.$sort)) {
       if (locationSettingSorts.includes(sort)) {
-        const data = dispatch.data ? dispatch.data : dispatch
+        const currentSort = context.params.query.$sort[sort]
+        delete context.params.query.$sort[sort]
 
-        data.sort((a, b) => {
-          let fa = a.locationSetting[sort],
-            fb = b.locationSetting[sort]
+        const query = context.service.createQuery(context.params)
 
-          if (typeof fa === 'string') {
-            fa = fa.toLowerCase()
-            fb = fb.toLowerCase()
-          }
+        query.join(locationSettingPath, `${locationSettingPath}.locationId`, `${locationPath}.id`)
+        query.orderBy(`${locationSettingPath}.${sort}`, currentSort === 1 ? 'asc' : 'desc')
+        query.select(`${locationPath}.*`)
+
+        context.params.knex = query
+      }
+    }
+  }
+}
+
+const addSearchParams = async (context: HookContext<LocationService>) => {
+  const { adminnedLocations, search } = context.params.query || {}
 
-          if (fa < fb) {
-            return -1
+  if (adminnedLocations && search) {
```

Comment:
`search` can be done by query on client side as done for `avatar` service https://github.com/EtherealEngine/etherealengine/pull/8973/files

Also, we can get rid of `adminnedLocations` and if its some how required it action be `action: admin` to make it standardized with other services.